src/dao/AccountDao.java

```diff
@@ -28,7 +29,7 @@ public AccountDao() {
      * @throws dao.DaoException if there is an error connecting to the database
      * or executing the query
      */
-    public ArrayList<Object> getAllAccounts() throws DaoException {
+    public ArrayList<Account> getAllAccounts() throws DaoException {
```

Comment:
We had this in the original design but I think we can simplify by removing it, unless you think there's a case in which we'd need to read more than one account at a time.